tests/keylkg.py

```diff
@@ -146,3 +146,12 @@ def test_GIVEN_running_ioc_WHEN_in_setup_mode_THEN_output1_switches_to_measureme
         self._lewis.backdoor_set_on_device("detector_1_raw_value", expected_value)
 
         self.ca.assert_that_pv_is("VALUE:OUTPUT:1", expected_value)
+
+    @parameterized.expand(parameterized_list(["Passive", "5 second"]))
+    @skip_if_recsim("Scan rate changes do not work in RECSIM.")
+    def test_WHEN_scan_rates_changed_THEN_scan_rates_set_correctly(self, _, rate):
```

Comment:
There is a test-ordering condition in these tests; you can test this by replacing

```
unittest.TestLoader().loadTestsFromNames(test_names)
```

with 

```
    loader = unittest.TestLoader()
    loader.sortTestMethodsUsing = lambda *a, **k: 1
    test_suite = loader.loadTestsFromNames(test_names)
```

in `run_tests.py`, this makes the tests run in a different order and then a load of them fail. The best fix for this is probably to set the `SCAN` back to `5 second` in `setUp()`.